server/tests/events.js

```diff
@@ -0,0 +1,399 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../app';
+import helper from './testHelper';
+
+const { expect } = chai;
+chai.use(chaiHttp);
+chai.should();
+
+module.exports = describe('Tests for Events', () => {
+  describe('POST api/v1/events', () => {
+    it('should return a status 403 error for no token', (done) => {
+      chai.request(app)
+        .post('/api/v1/events')
+        .send({
+          userId: 4,
+          centerId: 2,
+          eventType: 'Wedding',
+          email: helper.userEmail,
+          date: 5 / 12 / 2019,
+          guestNo: 1000
+        })
+        .end((err, res) => {
+          expect(res).to.be.status(403);
+          res.body.should.have.a('object');
+          res.body.should.have.property('message').to.equal('No token, please sign up or sign in');
+          done();
+        });
+    });
+
+    it('should return a status 401 error for authenticated token', (done) => {
+      chai.request(app)
+        .post(`/api/v1/events?token=${helper.wrongToken}`)
+        .send({
+          userId: 4,
+          centerId: 2,
+          eventType: 'Wedding',
+          email: helper.userEmail,
+          date: 5 / 12 / 2019,
+          guestNo: 1000
+        })
+        .end((err, res) => {
+          expect(res).to.be.status(401);
+          res.body.should.have.a('object');
+          res.body.should.have.property('message').to.equal('Failed to authenticate user');
+          done();
+        });
+    });
+
+    it('should return a status 400 error for an empty event type', (done) => {
+      chai.request(app)
+        .post(`/api/v1/events?token=${helper.userToken}`)
+        .send({
+          userId: 4,
+          centerId: 2,
+          eventType: '',
+          email: helper.userEmail,
+          date: 5 / 12 / 2019,
+          guestNo: 1000
+        })
+        .end((err, res) => {
+          expect(res).to.be.status(400);
+          res.body.should.have.a('object');
+          res.body.error[0].should.have.property('msg').to.equal('The event type cannot be empty');
+          done();
+        });
+    });
+
+    it('should return a status 400 error for a string input for the estimated guest number', (done) => {
+      chai.request(app)
+        .post(`/api/v1/events?token=${helper.userToken}`)
+        .send({
+          userId: 5,
+          centerId: 2,
+          eventType: 'Conference',
+          email: helper.userEmail,
+          date: 5 / 12 / 2019,
+          guestNo: 'one hundred'
+        })
+        .end((err, res) => {
+          expect(res).to.be.status(400);
+          res.body.should.have.a('object');
+          res.body.error[0].should.have.property('msg').to.equal('Your guest estimate must be a number more than 2');
+          done();
+        });
+    });
+
+    it('should return a status 400 error for too low estimated guest', (done) => {
```

Comment:
the comments about BDD and http status codes below also apply here.
so take note.